scylla/src/transport/load_balancing/mod.rs

```diff
@@ -62,7 +62,11 @@ pub type FallbackPlan<'a> = Box<dyn Iterator<Item = NodeRef<'a>> + Send + Sync +
 /// This trait is used to produce an iterator of nodes to contact for a given query.
 pub trait LoadBalancingPolicy: Send + Sync + std::fmt::Debug {
     /// Returns the first node to contact for a given query.
-    fn pick<'a>(&'a self, query: &'a RoutingInfo, cluster: &'a ClusterData) -> Option<NodeRef<'a>>;
+    fn pick<'a>(
+        &'a self,
+        query: &'a RoutingInfo,
+        cluster: &'a ClusterData,
+    ) -> Option<(NodeRef<'a>, Shard)>;
```

Comment:
I'm more for hiding `NodeRef`, `Shard` pair into some struct e.g.

```rust
pub struct PlanElement<'n> {
   node_ref: NodeRef<'n>,
   shard: Shard,
}
```

Converting `load_balancing` interfaces to use it instead of a plain tuple would allow us to have a greater flexibility in adding/removing `PlanElement`'s fields without breaking API.

Reply:
If we hide them, then how could a user implement their own policies???

Reply:
By using `PlanElement`s produced by the locator.

Reply:
What operations do you propose to be `pub` on `PlanElement` ? Will it be possible to examine its contents? Will it be possible to alter them? What about crafting one's own `PlanElement` ?

Reply:
> What operations do you propose to be pub on PlanElement ? Will it be possible to examine its contents?

Viewing `shard` and `node_ref` should be user-accessible.

> What about crafting one's own PlanElement ?

I don't immediately see why one would like to craft their own `PlanElement`. Objects required to do so (`NodeRef` + `Shard`) can only be obtained through locator.

> Will it be possible to alter them?

Allowing the user to influence a shard selection sounds like a good reason for altering the `shard` field.

Reply:
> I don't immediately see why one would like to craft their own PlanElement. Objects required to do so (NodeRef + Shard) can only be obtained through locator.
Precisely the case of #738 is one where a user would like to craft their own `PlanElement`, or at least this is a case close to crafting one's own.

Reply:
BTW, is `ReplicaLocator` only responsible for locating replicas or is it, from the LB policy point of view, also the only source of knowledge about non-necessarily-replica nodes in the cluster?